lib/main_page.dart

```diff
@@ -1,88 +1,189 @@
+                      itemCount: genres.length,
+                      itemBuilder: (context, index) => GenreBadge(
+                        genre: genres[index],
+                        isSelected: _selectedGenres[index],
+                        onTap: () {
+                          // for (int i = 0; i < _selectedGenres.length; i++) {
+                          //   _selectedGenres[i] = false;
+                          // }
+                          // _selectedGenres[index] = !_selectedGenres[index];
+                          // _filteredMovies = movies
+                          //     .where((movie) => movie.genre == genres[index])
+                          //     .toList();
+                          // setState(() {});
+                          _selectedGenres[index] = !_selectedGenres[index];
+                          if (_selectedGenres
+                              .every((element) => element == false)) {
+                            _filteredMovies = Set<Movie>.from(movies);
+                            setState(() {});
+                          } else {
+                            _filteredMovies = {};
+                            for (int i = 0; i < _selectedGenres.length; i++) {
+                              if (_selectedGenres[i]) {
+                                _filteredMovies.addAll(
+                                  movies
+                                      .where(
+                                        (movie) => movie.genre == genres[i],
+                                      )
+                                      .toList(),
+                                );
+                              }
+                              setState(() {});
+                            }
+                          }
+                        },
+                      ),
+                    ),
+                  ),
+                ],
+              ),
+            ),
+            Expanded(
+              child: ListView.separated(
+                separatorBuilder: (context, index) => const Divider(
+                  height: 1,
+                  color: Color.fromARGB(174, 86, 86, 86),
+                ),
+                itemCount: _filteredMovies.length,
+                itemBuilder: (context, index) => MovieCard(
+                  movie: _filteredMovies.toList()[index],
+                ),
+              ),
+            ),
+          ],
+        ),
+      );
+}
+
+class GenreBadge extends StatelessWidget {
+  const GenreBadge({
     super.key,
     required this.genre,
-    required this.onTap,
     required this.isSelected,
+    required this.onTap,
   });
 
   final Genre genre;
-  final VoidCallback onTap;
   final bool isSelected;
+  final VoidCallback onTap;
 
   @override
-  Widget build(context) => GestureDetector(
+  Widget build(BuildContext context) => GestureDetector(
         onTap: onTap,
-        child: Container(
-          height: 40,
+        child: Center(
           child: Padding(
```

Comment:
Let's remove outer padding from here. It should be added at the place where the badge is created.